modin/backends/pandas/query_compiler.py

```diff
@@ -2492,18 +2188,12 @@ def _dict_func(self, func, axis, *args, **kwargs):
         func = {idx: func[key] for key in func for idx in index.get_indexer_for([key])}
 
         def dict_apply_builder(df, func_dict={}):
-            return df.apply(func_dict, *args, **kwargs)
+            return pandas.DataFrame(df.apply(func_dict, *args, **kwargs))
```

Comment:
Why does this have to cast to a `pandas.DataFrame`?

Reply:
Sometimes `apply` can return a `Series`, but we require that internally all objects are `DataFrame`s.